### DISCUSSION TITLE
Raspberry Pi RP2040 Support?

### DISCUSSION THREAD

**Discussion by mgharrison** [Makuna/NeoPixelBus#537] [Q&A] [answered]
I'm trying to use NeoPixelBus on a Raspberry Pi Pico with a WS2811 light string.  I'm using PlatformIO as my development environment, and the information for the NeoPixelBus library lists "Raspberry Pi RP2040" as a compatible platform, so I thought this combo might work. I have the Raspberry Pi RP2040 version 1.5.0 platform installed.  However, when I try to compile any project with this library added I get a storm of compile errors, only a small representative sample is listed here because it mostly seems to be repeats of these same errors.
    In file included from .pio\libdeps\pico\NeoPixelBus\src/NeoPixelBus.h:118,
                     from src\main.cpp:14:
    .pio\libdeps\pico\NeoPixelBus\src/internal/NeoArmMethod.h:710:26: error: 'VARIANT_MCK' was not declared in this scope
      710 | #define ARM_OTHER_SCALE  VARIANT_MCK / 2UL / 1000000UL
          |                          ^~~~~~~~~~~
    .pio\libdeps\pico\NeoPixelBus\src/internal/NeoArmMethod.h:716:69: note: in expansion of macro 'ARM_OTHER_SCALE'
      716 |     static const uint32_t CyclesT0h = static_cast<uint32_t>((0.40 * ARM_OTHER_SCALE + 0.5) - (5 * ARM_OTHER_INST));
          |                                                                     ^~~~~~~~~~~~~~~
    .pio\libdeps\pico\NeoPixelBus\src/internal/NeoArmMethod.h:711:33: error: 'F_CPU' was not declared in this scope

It may well be that I'm just trying to do something that is not supposed to work, but I wonder, is there something I'm likely doing wrong here and I can work around this?

**Comment by Makuna** [comment, marked as answer]
Do you have a link as to where it states this library supports Raspberry Pi? 

Those errors are from missing compiler symbols.  Your board/build environment isn't including all the standard Arduino symbols like F_CPU.  They are usually supplied as command line parameters during the build (all this is hidden in Arduino world).

The specific Arduino RP2040 board support using the Arduino IDE is not supported.  Currently `ARDUINO_ARCH_RP2040` and `ARDUINO_ARCH_MBED_NANO` is what this board uses, which are not directly supported, so it falls into the generic ARM support in my library, in which this board fails to define all the normal ARM Arduino symbols.  Talk to the Arduino guys about this.

**Reply by mgharrison** [reply]
I don't have a link, but this is how it shows up in the Libraries listing in PIO:
![image](https://user-images.githubusercontent.com/5024193/144723575-a48c79a9-099b-45a2-948b-0325d1e90c6a.png)

**Reply by Makuna** [reply]
I don't directly support PIO, so I have no clue where that text comes from, but it doesn't come from my library.  Maybe ask on a PIO forum to see if someone had gotten it work and thus the added text.
